qiskit/synthesis/multi_controlled/mcx_synthesis.py

```diff
@@ -122,8 +131,8 @@ def synth_mcx_n_dirty_i15(
     return qc
 
 
-def synth_mcx_n_clean_m15(num_ctrl_qubits: int):
-    """
+def synth_mcx_n_clean_m15(num_ctrl_qubits: int) -> QuantumCircuit:
```

Comment:
please note that in this code we also provide the circuit names (this is relevant to all the functions).
these are currently "mcx_vchain" or "mcx_recursive".
I would suggest to change them to "mcx" and update the QASM tests accordingly (if it's not considered breaking API)

Reply:
We should be careful with the name `"mcx"`, for example, I could imagine that calling the `AerSimulator` with a decomposition with ancillas and the name `"mcx"` will cause wrong results, as the simulator could think that we have more controls.

Wouldn't it be safer to just call them individually (i.e. `mcx_m15`)? If we reach this function, then the circuit is getting synthesized, so I think it's fine for the circuit to change and for operations to change names.

Reply:
I actually did this in one of the previous commits and I believe it broke some of the QASM tests: for instance `synth_mcx_n_clean_m15` is used in the definition of `MCXVChain`. We will be able to clean it up after the custom MCX gates are deprecated. Though please note that the name "mcx" is not ideal either, since we might want to eventually enforce that an object called "mcx" is an "MCXGate" (the same way as an object called "h'" is a Hadamard); this is already not the case (and requires additional `isinstance` checks in the plugins), but let's not make the situation worse.